skill/dialog.py

```diff
@@ -0,0 +1,401 @@
+        )
+        self._add_location()
+
+    def build_condition_dialog(self, intent_match: bool):
+        """Select the relevant dialog file for condition based reports.
+
+        A condition can for example be "snow" or "rain".
+
+        :param intent_match: true if intent matches a vocabulary for the condition
+        """
+        self.data = dict(
+            condition=self.weather.condition.description.lower(),
+            day=get_speakable_day_of_week(self.weather.date_time),
+        )
+        if intent_match:
+            self.name += ".condition.expected"
+        else:
+            self.name += ".condition.not.expected".format(
+                self.weather.condition.category.lower()
+            )
+        self._add_location()
+
+    def build_sunrise_dialog(self):
+        """Build the components necessary to speak the sunrise time."""
+        self.name += ".sunrise"
+        self.data = dict(time=nice_time(self.weather.sunrise))
+        self.data.update(day=get_speakable_day_of_week(self.weather.date_time))
+        self._add_location()
+
+    def build_sunset_dialog(self):
+        """Build the components necessary to speak the sunset time."""
+        self.name += ".sunset"
+        self.data = dict(time=nice_time(self.weather.sunset))
+        self.data.update(day=get_speakable_day_of_week(self.weather.date_time))
+        self._add_location()
+
+
+class WeeklyDialog(Dialog):
+    """Weather dialog builder for weekly weather."""
+
+    def __init__(
+        self,
+        intent_data: WeatherIntent,
+        config: WeatherConfig,
+        forecast: List[DailyWeather],
+    ):
+        super().__init__(intent_data, config)
+        self.forecast = forecast
+        self.name = "weekly"
+
+    def build_temperature_dialog(self):
+        """Build the components necessary to temperature ranges for a week."""
+        low_temperatures = [daily.temperature.low for daily in self.forecast]
+        high_temperatures = [daily.temperature.low for daily in self.forecast]
+        self.name += ".temperature"
+        self.data = dict(
+            low_min=min(low_temperatures),
+            low_max=max(low_temperatures),
+            high_min=min(high_temperatures),
+            high_max=max(high_temperatures),
+        )
+
+    def build_condition_dialog(self, condition: str):
+        """Build the components necessary to speak the days of week for a condition."""
+        self.name += ".condition"
+        self.data = dict(condition=condition)
+        days_with_condition = []
+        for daily in self.forecast:
+            if daily.condition.category == condition:
+                day = get_speakable_day_of_week(daily.date_time)
+                days_with_condition.append(day)
+        if len(days_with_condition) > 1:
+            last_day = days_with_condition.pop(-1)
+            days_with_condition.append("and " + last_day)
+        self.data.update(days=", ".join(days_with_condition))
```

Comment:
but if there are 4 days, the response will be "day one and day two and day three and day four".

Reply:
It looks like this code will output: "day one, day two, day three, and day four"

This is also what `join_list` does - only uses the "and" for the final item in the list. We added it because the same code was in heaps of Skills that were reading out lists of options.

Reply:
Oh.  I misread your first comment.  Didn't know that function existed.  Will change.